perceval/runtime/remote_backend.py

```diff
@@ -28,118 +28,89 @@
 
 from perceval.backends import Backend
 from perceval.components import ACircuit
+from .rpc_handler import RPCHandler
 from perceval.serialization import serialize, bytes_to_jsonstring, deserialize_state, deserialize_state_list, \
     deserialize_float
-from perceval.utils import Matrix, BasicState
+from perceval.utils import Matrix, BasicState, generate_sync_methods
 
 from pkg_resources import get_distribution
 
 pcvl_version = get_distribution("perceval-quandela").version
-JOB_CREATE_ENDPOINT = '/api/job'
-
-
-################################
-# Sync methods Factory
-def _sync_wrapper(cls, func):
-    async_func = getattr(cls, func)
-
-    def await_job(*args):
-        job = async_func(*args)
-        while True:
-            if not job.is_completed():
-                time.sleep(3)
-            else:
-                return job.get_results()
-
-    return await_job
-
-
-def generate_sync_methods(cls):
-    for method in dir(cls):
-        if method.startswith('async_'):
-            sync_name = method.removeprefix('async_')
-            setattr(cls, sync_name, _sync_wrapper(cls, method))
-    return cls
 
 
 ################################
 @generate_sync_methods
 class RemoteBackend(Backend):
```

Comment:
The payloads we need to generate do not follow the Backend interface. As such, I thought we said the RemoteBackend class should disappear and its responsibility (be able to generate payloads) transferred to the RemoteProcessor (which actually follows the API we need to expose - samples(), sample_count() and probs())

Reply:
Also, the payload should contain Processor._parameters key/value pairs, so it makes even more sense to give the responsibility of creating payloads to the RemoteProcessor.

For instance, the noise parameter, expected by some simulators, is meant to be in Processor._parameters dict.